helm/prometheus-rules/templates/alerting-rules/teleport.rules.yml

```diff
@@ -16,17 +16,11 @@ spec:
       expr: |
         count (
           kube_secret_created{secret=~".*-teleport-join-token"}
-        ) by (cluster_id, installation)
```

Comment:
Can you add the by (cluster_id, installation) back ? Those are needed for milir

Reply:
Why are they différent ?

Reply:
That's why we had the relabel in place in the second expression

Reply:
By summing them by cluster_id/installation you have 1 cluster of each 
```
        sum (
          label_replace(
            capi_cluster_status_phase{phase="Provisioned"},
            "cluster_id",
            "$1",
            "name",
            "(.*)"
          )
        ) by (cluster_id, installation)
```
![image](https://github.com/giantswarm/prometheus-rules/assets/138458697/2697fe43-9b0d-4e5c-bf2b-802ad3c150cb)

and we need the total of them to compare with 
```
        count (
          kube_configmap_info{configmap=~".*-teleport-kube-agent-config"}
        ) by (cluster_id, installation)
```
![image](https://github.com/giantswarm/prometheus-rules/assets/138458697/43cd1b1e-6474-40ab-98a5-9fd8b1e0d3cc)

Reply:
We can can keep the label_replace but we have to get rid of the `by (cluster_id, installation)` part

Reply:
or we can add an additional sum on top

Reply:
You might need to use something like this :( https://github.com/giantswarm/prometheus-rules/blob/3e1587b2ee57abcd98dbf07e193b50b481444d23/helm/prometheus-rules/templates/recording-rules/grafana-cloud.rules.yml#L8

unless @hervenicol can think of a better solution?

Reply:
After some testing on grizzly, I would replace the configmap part with:

```promql
count (
    label_replace(
      kube_configmap_info{configmap=~".*-teleport-kube-agent-config"}, 
      "cluster_id",
      "$1",
      "configmap", 
      "(.*)-teleport-kube-agent-config"
    )
) by (cluster_id, installation, provider, pipeline) 
!= sum (
    capi_cluster_status_phase{phase="Provisioned"}
) by (cluster_id, installation, provider, pipeline)
```

Reply:
and the secret one with:


```promql
count (
    label_replace(
      kube_secret_created{secret=~".*-teleport-join-token"}, 
      "cluster_id",
      "$1",
      "secret", 
      "(.*)-teleport-join-token"
    )
) by (cluster_id, installation, provider, pipeline)
!= sum (
    capi_cluster_status_phase{phase="Provisioned"}
) by (cluster_id, installation, provider, pipeline)
```

Reply:
nice! LGTM

Reply:
updating the queried on rules